apps/outlook-rooms-addin/src/tests/room-booking.component.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { RoomBookingComponent } from '../app/rooms/room-booking.component';
+import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
+import {
+    FormsModule,
+    ReactiveFormsModule,
+    FormBuilder,
+    Validators,
+} from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { BookModule } from '../app/rooms/book.module';
+import { Router } from '@angular/router';
+import { EventFormService } from '@placeos/events';
+import { mockForm, mockRouterStub } from './test-mocks';
+
+describe('RoomBookingComponent', () => {
+    const formModel = mockForm;
+    const fb = new FormBuilder();
+    const form = fb.group(formModel);
+
+    const RouterStub = mockRouterStub;
+    let spectator: Spectator<RoomBookingComponent>;
+
+    const createComponent = createComponentFactory({
+        component: RoomBookingComponent,
+        imports: [
+            ReactiveFormsModule,
+            FormsModule,
+            MatFormFieldModule,
+            BookModule,
+        ],
+        providers: [
+            {
+                provide: Router,
+                useClass: RouterStub,
+            },
+            {
+                provide: EventFormService,
+                useValue: {
+                    setView: jest.fn(() => {}),
+                    newForm: jest.fn(() => {}),
+                    clearForm: jest.fn(),
+                    storeForm: jest.fn(() => {}),
+                    loadForm: jest.fn(),
+                    postForm: jest.fn(),
+                    view: '',
+                },
+            },
+        ],
+        declarations: [],
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+        const event_service: any = spectator.inject(EventFormService);
+        event_service.setView.mockImplementation((_) => {
+            event_service.view = _;
+            spectator.detectChanges();
+        });
+        event_service.newForm.mockImplementation((_) => {
+            event_service.form = fb.group(formModel);
+        });
+        event_service.storeForm.mockImplementation((_) => {
+            if (
+                event_service.form.controls.title == null ||
```

Comment:
You should use the `contains` method on the form group for this instead.
e.g. `event_service.form.contains('title')`